src/main/java/io/nats/client/impl/Headers.java

```diff
@@ -390,6 +394,40 @@ public ByteArrayBuilder appendSerialized(ByteArrayBuilder bab) {
 		return bab;
 	}
 
+	/**
+	 * Write the header to the byte array. Assumes that the caller has
+	 * already validated that the destination array is large enough by using getSerialized()
+	 * @param destPosition the position index in destination byte array to start
+	 * @param dest the byte array to write to
+	 * @return the length of the header
+	 */
+	public int serializeToArray(int destPosition, byte[] dest) {
+		System.arraycopy(HEADER_VERSION_BYTES_PLUS_CRLF, 0, dest, destPosition, HVCRLF_BYTES);
+		destPosition += HVCRLF_BYTES;
+
+		for (Map.Entry<String, List<String>> entry : valuesMap.entrySet()) {
+			List<String> values = entry.getValue();
+			for (String value : values) {
+				byte[] bytes = entry.getKey().getBytes(US_ASCII);
+				System.arraycopy(bytes, 0, dest, destPosition, bytes.length);
+				destPosition += bytes.length;
+
+				dest[destPosition++] = COLON;
+
+				bytes = value.getBytes(US_ASCII);
+				System.arraycopy(bytes, 0, dest, destPosition, bytes.length);
+				destPosition += bytes.length;
+
+				dest[destPosition++] = CR;
+				dest[destPosition++] = LF;
+			}
+		}
+		dest[destPosition++] = CR;
```

Comment:
I'm wondering if these CRLF can/should be removed.

Looking in the [`NatsConnectionWriter`](https://github.com/nats-io/nats.java/pull/776/files#diff-7b435bb1d33408d9aa01983b0c43086b0da7f479636890d2f05d248b0971f676L139), it looks like the extra CRLF was not present before?